s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -43,6 +41,13 @@ final case class FailedUpload(reasons: Seq[Throwable]) extends Exception
 
 final case class CompleteMultipartUploadResult(location: Uri, bucket: String, key: String, etag: String)
 
+final case class ListBucketResult(isTruncated: Boolean, continuationToken: Option[String], keys: Seq[String])
+
+sealed trait ListBucketState
```

Comment:
Can you move those into a private place so that it doesn't clutter the public namespace? Would be ok from my side to but the whole hierarchy into the `listBucket` method if it's only used there.